crates/swc_ecma_transforms_typescript/src/inline_enum.rs

```diff
@@ -0,0 +1,150 @@
+use std::{cell::RefCell, rc::Rc};
+
+use swc_atoms::JsWord;
+use swc_common::collections::AHashMap;
+use swc_ecma_ast::{
+    AssignExpr, AssignPat, AssignPatProp, ComputedPropName, Expr, Id, Lit, MemberExpr, MemberProp,
+    Module, Script, Str, Tpl,
+};
+use swc_ecma_utils::ident::IdentLike;
+use swc_ecma_visit::{as_folder, noop_visit_mut_type, Fold, VisitMut, VisitMutWith};
+
+use crate::TSEnumConfig;
+
+pub(crate) type TSEnumLit = Rc<RefCell<AHashMap<Id, AHashMap<JsWord, Lit>>>>;
+
+pub fn inline_enum(ts_enum_lit: TSEnumLit, ts_enum_config: TSEnumConfig) -> impl Fold + VisitMut {
+    as_folder(InlineEnum {
+        ts_enum_lit,
+        ts_enum_config,
+        is_lhs: false,
+    })
+}
+
+struct InlineEnum {
+    ts_enum_lit: TSEnumLit,
+    ts_enum_config: TSEnumConfig,
+
+    is_lhs: bool,
+}
+
+trait SkipInlineEnum {
+    fn skip(&self) -> bool;
+}
+
+impl SkipInlineEnum for InlineEnum {
```

Comment:
Seems like this is the only impl.
I think it can be a method of `InlineEnum`

Reply:
I had implemented this method on another pub struct while I want to hide it, so I wrote this trait.
I am going to fix it.